Content.Server/Body/Systems/BloodstreamSystem.cs

```diff
@@ -164,6 +164,19 @@ private void OnBeingGibbed(EntityUid uid, BloodstreamComponent component, BeingG
         SpillAllSolutions(uid, component);
     }
 
+    private void OnApplyMetabolicMultiplier(EntityUid uid, BloodstreamComponent component, ApplyMetabolicMultiplierEvent args)
+    {
+        if (args.Apply)
+        {
+            component.UpdateInterval *= args.Multiplier;
+            return;
+        }
+        component.UpdateInterval /= args.Multiplier;
+        // Reset the accumulator properly
+        if (component.AccumulatedFrametime >= component.UpdateInterval)
+            component.AccumulatedFrametime = component.UpdateInterval;
+    }
```

Comment:
Just applying it every time is gonna cause problems if the event is ever called again. It should ideally reset.

Reply:
That was how I did it before it was made generic, I changed it because otherwise multiple sources of this effect would break everything

Reply:
this is probably fine